fusequery/query/src/interpreters/interpreter_describe_table.rs

```diff
@@ -0,0 +1,70 @@
+// Copyright 2020-2021 The Datafuse Authors.
+//
+// SPDX-License-Identifier: Apache-2.0.
+
+use std::sync::Arc;
+
+use common_datablocks::DataBlock;
+use common_datavalues::prelude::*;
+use common_datavalues::series::Series;
+use common_exception::Result;
+use common_planners::DescribeTablePlan;
+use common_streams::DataBlockStream;
+use common_streams::SendableDataBlockStream;
+
+use crate::interpreters::Interpreter;
+use crate::interpreters::InterpreterPtr;
+use crate::sessions::FuseQueryContextRef;
+
+pub struct DescribeTableInterpreter {
+    ctx: FuseQueryContextRef,
+    plan: DescribeTablePlan,
+}
+
+impl DescribeTableInterpreter {
+    pub fn try_create(ctx: FuseQueryContextRef, plan: DescribeTablePlan) -> Result<InterpreterPtr> {
+        Ok(Arc::new(DescribeTableInterpreter { ctx, plan }))
+    }
+}
+
+#[async_trait::async_trait]
+impl Interpreter for DescribeTableInterpreter {
+    fn name(&self) -> &str {
+        "DescribeTableInterpreter"
+    }
+
+    async fn execute(&self) -> Result<SendableDataBlockStream> {
+        let datasource = self.ctx.get_datasource();
```

Comment:
maybe `self.ctx.get_table(db, table)` is better ?

Reply:
Good suggestion, done.